### PR TITLE
Added a solution to q.12.1 - added a check to eliminate empty string which are pushed using getline

### PR DESCRIPTION
**Pull request by kohn1001**
…k to eliminate empty strings
Without this check it actually doesn't print the last k lines - (tested on Mac)

### CODE REVIEW

**Comment by kohn1001** [comment]
Much better version - building a vector of size K (in case K is much larger then the number of lines in file - it's much more efficient)

File: Chapter-16-moderate/word_freq.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ *
+*/
+
+int countWordFreq(const string & fileName, const string &word)
+{
+	cout << "file path is: " << fileName << endl;
+	ifstream ifn(fileName);
+	static map<string, int> wordFreq;
+	string tword;
+	transform(word.begin(), word.end(), tword.begin(), ::tolower);
```
Comment (sergio-nsk) [inline Chapter-16-moderate/word_freq.cpp:33]: Transform to the empty target destination crashes the app.

File: Ch 1.Arrays And Strings/3.URLify/URLify2.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
+#include <string>
+
+#include <iostream>
+using namespace std;
```
Comment (sergio-nsk) [inline Ch 1.Arrays And Strings/3.URLify/URLify2.cpp:4]: Very bad habit.
[Why is "using namespace std;" considered bad practice?](https://stackoverflow.com/questions/1452721/why-is-using-namespace-std-considered-bad-practice)

File: Ch 12. C++/print_k_lines.cpp [hunk 3]
```diff
@@ -0,0 +1,48 @@
+#include <iostream>
+
+
+#include <string>
+#include <vector>
+#include <fstream>
+using namespace std;
```
Comment (sergio-nsk) [inline Ch 12. C++/print_k_lines.cpp:7]: Very bad habit.
[Why is "using namespace std;" considered bad practice?](https://stackoverflow.com/questions/1452721/why-is-using-namespace-std-considered-bad-practice)

File: Chapter-16-moderate/word_freq.cpp [hunk 4]
```diff
@@ -0,0 +1,66 @@
+#include <iostream>
+#include <string>
+#include <fstream>
+#include <sstream>
+#include <map>
+#include <cctype>
+#include <algorithm>
+
+using namespace std;
```
Comment (sergio-nsk) [inline Chapter-16-moderate/word_freq.cpp:9]: Very bad habit.
[Why is "using namespace std;" considered bad practice?](https://stackoverflow.com/questions/1452721/why-is-using-namespace-std-considered-bad-practice)

File: Chapter-16-moderate/word_freq.cpp [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+int countWordFreq(const string & fileName, const string &word)
+{
+	cout << "file path is: " << fileName << endl;
+	ifstream ifn(fileName);
+	static map<string, int> wordFreq;
+	string tword;
+	transform(word.begin(), word.end(), tword.begin(), ::tolower);
+
+	if (!wordFreq.empty()) {
```
Comment (sergio-nsk) [inline Chapter-16-moderate/word_freq.cpp:35]: Redundant if just after created empty `wordFreq`.

File: Chapter-16-moderate/word_freq.cpp [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+		if(line.empty()) { continue; }
+		stringstream ss(line);
+		string w;
+		while(ss >> w) {
+			transform(w.begin(), w.end(), w.begin(), ::tolower);
+			wordFreq[w]++;
+		}
+	}
+
+	return wordFreq[word];
```
Comment (sergio-nsk) [inline Chapter-16-moderate/word_freq.cpp:52]: You placed lowercased words into the map, `word` can has capital letters.

**Review by sergio-nsk — CHANGES_REQUESTED** [review]

**Comment by sergio-nsk** [comment]
@KrishMunot Why have you approved and merged the PR containing many bugs? Especially the crash/undefined behavior by [`std::transform()` to the empty container](https://github.com/careercup/CtCI-6th-Edition-cpp/pull/32#discussion_r961317756)?

**Comment by KrishMunot** [comment]
My bad, feel free to revert it, i thought some of the commits were valid.
Please revert it.

**Comment by sergio-nsk** [comment]
I can revert it, but it will require your approve, since I have not right to merge.

**Comment by KrishMunot** [comment]
Sure, no problem. I'll approve it.

**Pull request merged**